projects/core/src/site-context/config/config-loader/site-context-config-loader.service.spec.ts

```diff
@@ -0,0 +1,133 @@
+import { PLATFORM_ID } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { SERVER_REQUEST_URL } from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { BaseSite } from '../../../model/misc.model';
+import { BaseSiteService } from '../../facade/base-site.service';
+import { SiteContextConfigLoaderService } from './site-context-config-loader.service';
+
+describe(`SiteContextConfigLoaderService`, () => {
+  let service: SiteContextConfigLoaderService;
+  let baseSiteService: BaseSiteService;
+
+  let mockBaseSites: BaseSite[];
+  const mockServerRequestUrl = 'test-server-request-url';
+
+  function beforeEachWith({ platform }: { platform: string }) {
+    const mockBaseStore = {
+      languages: [{ isocode: 'de' }, { isocode: 'en' }],
+      defaultLanguage: { isocode: 'en' },
+      currencies: [{ isocode: 'EUR' }, { isocode: 'USD' }],
+      defaultCurrency: { isocode: 'EUR' },
+    };
+
+    mockBaseSites = [
+      {
+        uid: 'test',
+        urlPatterns: ['testUrl'],
+        baseStore: mockBaseStore,
+        urlEncodingAttributes: ['language', 'currency'],
+        theme: 'test-theme',
+      },
+    ];
+
+    class MockBaseSiteService {
+      getAll(): Observable<BaseSite> {
+        return of({});
+      }
+    }
+
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: BaseSiteService, useClass: MockBaseSiteService },
+        { provide: SERVER_REQUEST_URL, useValue: mockServerRequestUrl },
+        { provide: PLATFORM_ID, useValue: platform },
+      ],
+    });
+
+    service = TestBed.inject(SiteContextConfigLoaderService);
+    baseSiteService = TestBed.inject(BaseSiteService);
+  }
+
+  describe(`loadConfig - should return config based on the base site data`, () => {
+    describe(`on BOWSER platform`, async () => {
+      beforeEach(() => beforeEachWith({ platform: 'browser' }));
+
+      it(`should throw error when the base sites loaded are undefined`, async () => {
+        spyOn(baseSiteService, 'getAll').and.returnValue(of(undefined));
+        let message = false;
+        try {
+          await service.loadConfig();
+        } catch (e) {
+          message = e.message;
+        }
+        expect(message).toBeTruthy();
```

Comment:
Bagatelle: Did you consider using `expect(()=> .... ).toThrow()` for tests in this file?

Reply:
I thought I tried this `expect(()=> .... ).toThrow()` before, but I does not pass. So, I changed it to this `expect(message).toBeTruthy();`.

Reply:
Perhaps it needed `await` inside the arrow function. I'll check it